tests/python_package_test/test_engine.py

```diff
@@ -3046,6 +3047,26 @@ def test_interaction_constraints():
                     train_data, num_boost_round=10)
 
 
+def test_linear_trees_num_threads(tmp_path):
+    # check that number of threads does not affect result
+    fd = FileLoader(EXAMPLES_DIR / 'binary_classification', 'binary',
+                    'train_linear.conf')
```

Comment:
I've changed it to use some randomly generated data (I was unable to reproduce the failure with the breast cancer dataset).

Reply:
Ah ok, I didn't realize that it could be dataset-specific. Sorry for creating extra work for you! I guess we could have also avoided "test file importing another test file" by moving `FileLoader` to `utils.py`. That would probably good to do *anyway* (in a separate PR).

Anyway, if the new randomly-generated data is sufficient to reproduce the underlying issue fixed by this PR, I'm good with it!